### ISSUE TITLE
An error "Error response from daemon..." occurs when checking images/containers in Windows Container

### ISSUE THREAD

**Issue by v-ruizh** [docker/for-win#13058]
**OS:** Win10
**Docker Desktop:** 4.14.0

**More Info:**
1. This issue does not reproduce in Linux Container.
2. This issue can be easily reproduced in new VM.
3. Docker Desktop is running without any error after switching to Windows Container.
4. This issue is unstable in existing VMs.

**Repro Steps:**
1. Install Docker Desktop 4.14.0 in a new Windows 10 VM.
2. Start Docker Desktop.
5. Click "Switch to Windows Containers...".
6. Use command "docker images" or "docker ps" to check the images/containers.

**Expect:**
There is no error.

**Actual:**
An error "Error response from daemon..." occurs.
![image](https://user-images.githubusercontent.com/45614674/200813833-5ddac625-aa82-418e-8e63-bf8806339a6a.png)

**Comment by sirinsidiator** [comment]
I'm having the same issue on Windows 11 Pro 22H2 (22621.819) witch Docker v4.14.0. I was trying to run a windows image and for that purpose switched to windows containers.
`docker info` gives me the following output:
```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.9.1)
  compose: Docker Compose (Docker Inc., v2.12.2)
  dev: Docker Dev Environments (Docker Inc., v0.0.3)
  extension: Manages Docker extensions (Docker Inc., v0.2.13)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.21.0)

Server:
ERROR: Error response from daemon: open \\.\pipe\docker_engine_windows: The system cannot find the file specified.
errors pretty printing info
```
`com.docker.diagnose.exe check` returns this:
```
[2022-11-12T11:52:17.772014900Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
Error response from daemon: open \\.\pipe\docker_engine_windows: The system cannot find the file specified.
[FAIL] DD0003: is the Docker CLI working? exit status 1
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?

Please investigate the following 1 issue:

1 : The test: is the Docker CLI working?
    Failed with: exit status 1

The Docker CLI is needed to build, share and run applications.
```

Following some google search results I tried to completely uninstall and reinstall Docker for Windows, install all Windows Features for Hyper-V and double checked it is running without errors via the event viewer following [these steps](https://superuser.com/questions/1026651/how-to-find-out-whether-hyper-v-is-currently-enabled-running).
WSL2 images work fine for me. It's just the windows images I can not run.

**Comment by sirinsidiator** [comment]
Seems the Hyper-V feature in the windows dialogue does not include the thing Docker needs. After reverting to v4.13.1 I got the following error when switching to windows containers:
![Error](https://user-images.githubusercontent.com/10908027/201475029-336a4ade-861c-490b-8e55-d1b143d987e9.png)
Running `Enable-WindowsOptionalFeature -Online -FeatureName $("Microsoft-Hyper-V", "Containers") -All` and restarting seems to have fixed it, even after updating back to v4.14.0. No idea why the error was not showing in 4.14.0 though.
